### DISCUSSION TITLE
EEBUS as home Assistant Integration

### DISCUSSION THREAD

**Discussion by mgiesl2s** [enbility/eebus-go#91] [Ideas]
Did you think to make a
EEBUS integration in Home Assistant?
I use a Vaillant Arotherm plus heating pump and want to maximaze the use of Solar power at daytime and to prevent cycling.

A Home Assistant integration where you can make an Automation will be cool :-)

**Comment by DerAndereAndi** [comment]
Hi,

thanks for your question. This repository is providing a protocol implementation for EEBUS and some of its use cases. So it is the foundation to implement support into various systems, but can not be integrated as is e.g. into Home Assistant. Also it is written in Go, so non-Go systems need to implement an interface they can communicate with.

The Vaillant Arotherm heating pump uses UseCases that are currently not supported by this EEBUS stack implementation. From the data I got so far, it is also not clear who this can work.

So at the end there are two missing items:
- UseCase implementations in this repository for what the specific heat pump uses 
- Another service a system like Home Assistant can use to communicate

In general: I currently can not say what kind of functionality the EEBUS use cases used in the heat pump will expose. It definitely will not expose individual values like temperatures and the option to change them.

**Comment by deanjesper** [comment]
I've just had a Vaillant ASHP installed and I'm gutted to read that there's no way of getting any useful data out of it over EEBUS.

I've read that SolarEdge and Vaillant have integrated with EEBUS. Would it be possible to reverse engineer that?

**Reply by DerAndereAndi** [reply]
There is nothing that needs to be reverse engineered. All the usecases with the data points and functional possibilities are defined in publicly available specifications at eebus.org.
